src/corenetwork/lteip/IP2lte.cc

```diff
@@ -42,44 +64,47 @@ void IP2lte::initialize(int stage)
         ueHold_ = false;
 
         binder_ = getBinder();
-
-        if (nodeType_ == ENODEB)
-        {
-            // TODO not so elegant
+        
+        if (nodeType_ == ENODEB) {
             cModule *enodeb = getParentModule()->getParentModule();
             MacNodeId cellId = getBinder()->registerNode(enodeb, nodeType_);
             nodeId_ = cellId;
-            registerInterface();
         }
     }
-    if (stage == inet::INITSTAGE_NETWORK_LAYER - 1)  // the configurator runs at stage NETWORK_LAYER, so the interface
-    {                                                // must be configured at a previous stage
-        if (nodeType_ == UE)
-        {
-            // TODO not so elegant
+    else if (stage == inet::INITSTAGE_LINK_LAYER) {
+        if(nodeType_ == ENODEB) {
+            registerInterface();
+        } else if (nodeType_ == UE) {
             cModule *ue = getParentModule()->getParentModule();
             nodeId_ = binder_->registerNode(ue, nodeType_, ue->par("masterId"));
             registerInterface();
-
+        } else {
+            throw cRuntimeError("unhandled node type: %i", nodeType_);
+        }
+    }
+    else if (stage == inet::INITSTAGE_NETWORK_LAYER - 1) {
```

Comment:
Hmm, the -1 is a bad idea, you can't rely on that. What is the intention here?

Reply:
This is a left-over from the INET3 version, I guess the intention was to have the statements executed before the network stage. Replaced it with the right init stages.